src/stores/WalletStore.ts

```diff
@@ -102,16 +116,45 @@ export default class WalletStore {
 
   @action
   public logout = () => {
-    this.app.walletStore.stopPolling();
+    this.reset();
     this.app.routerStore.push('/login');
   }
 
+  @action
+  public reset = () => {
+    this.stopPolling();
+    this.info =  INIT_VALUES.info;
+    this.loggedInAccount = INIT_VALUES.loggedInAccount;
+    this.wallet = INIT_VALUES.wallet;
+  }
+
+  @action
+  public resetWithNetwork = () => {
+    this.reset();
+    this.accounts = INIT_VALUES.accounts;
+  }
+
   @action
   private recoverWallet(mnemonic: string) {
-    const network = networks.testnet;
+    const network = this.app.networkStore.network;
     this.wallet = network.fromMnemonic(mnemonic);
   }
 
+  private handleGetChromeStorageReturn(storageAccounts: any[]) {
+    // Account not found, route to Create Wallet page
+    if (isEmpty(storageAccounts)) {
+      this.app.routerStore.push('/create-wallet');
+      this.loading = false;
+      return;
+    }
+    // Accounts found, route to Login page
+    this.accounts = storageAccounts;
+     // This is used to set the default selected account on the login page, we also call it indirectly in Login/index.tsx componentDidMount so that the default account is set when a user logs out (without switching networks), which will not hit this chrome storage flow
+    this.app.loginStore.selectedWalletName = this.accounts[0].name;
```

Comment:
better to let the component handle this. is there a reason why you need to set this here? as long as the accounts are set before hitting the login page then it should be fine.

Reply:
when you say let the component handle this, are you referring to doing so in the login components componentDidMount? handling it there is fine if the page is being reloaded, but if someone switches the network while already on that page, it won't hit the componentDidMount lifecycle hook. Ideally I would let the login component handle it based on some observable(such as the networkIndex  changing in the networkStore), but I also need to wait until the account has finished loading from chrome storage before doing so. The only way I thought to do that would have been to include a flag here for networkSwitchingInProcess that if toggled here would trigger the reaction on the login component page to reload selectedWalletName from the accounts. I explored that, and it worked, but it seemed more complicated then just including the single line here. Do you think that is better? or do you have another suggestion for how to go about it?

Reply:
how about create a reaction in `LoginStore`:
```
reaction(
  () => this.app.walletStore.accounts,
  () => this.selectedWalletName = this.app.walletStore.accounts[0].name;
```